File: backend-ms1/src/main/java/ma/zs/stocky/service/facade/admin/commun/MarqueAdminService.java
package ma.zs.stocky.service.facade.admin.commun;

import java.util.List;
import ma.zs.stocky.bean.core.commun.Marque;
import ma.zs.stocky.dao.criteria.core.commun.MarqueCriteria;
import ma.zs.stocky.zynerator.service.IService;

import ma.zs.stocky.ws.dto.commun.MarqueDto;
import org.springframework.http.HttpEntity;


public interface MarqueAdminService extends  IService<Marque,MarqueCriteria>  {



    HttpEntity<byte[]> createPdf(MarqueDto dto) throws Exception;

}

File: frontend/src/app/controller/service/admin/commun/MarqueAdmin.service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {BehaviorSubject, Observable} from 'rxjs';

import { RoleService } from 'src/app/zynerator/security/Role.service';
import {environment} from 'src/environments/environment';

import {MarqueDto} from 'src/app/controller/model/commun/Marque.model';
import {MarqueCriteria} from 'src/app/controller/criteria/commun/MarqueCriteria.model';
import {AbstractService} from 'src/app/zynerator/service/AbstractService';


@Injectable({
  providedIn: 'root'
})
export class MarqueAdminService extends AbstractService<MarqueDto, MarqueCriteria> {
     constructor(private http: HttpClient, private roleService: RoleService) {
        super();
        this.setHttp(http);
        this.setApi(environment.apiUrl + 'admin/marque/');
    }

    public constrcutDto(): MarqueDto {
        return new MarqueDto();
    }

    public constrcutCriteria(): MarqueCriteria {
        return new MarqueCriteria();
    }
}
